packages/ckeditor5-image/tests/imageresizeui.js

```diff
@@ -0,0 +1,220 @@
+import ImageResizeUI from '../src/imageresizeui';
+import ImageStyle from '../src/imagestyle';
+import Undo from '@ckeditor/ckeditor5-undo/src/undo';
+import Table from '@ckeditor/ckeditor5-table/src/table';
+
+import testUtils from '@ckeditor/ckeditor5-core/tests/_utils/utils';
+
+describe( 'ImageResizeUI', () => {
+	let plugin, command, editor, editorElement;
+	const resizeOptions = [ {
+		name: 'imageResize:original',
+		label: 'Original size',
+		value: null
+	},
+	{
+		name: 'imageResize:50',
+		label: '50%',
+		value: '50'
+	},
+	{
+		name: 'imageResize:75',
+		label: '75%',
+		value: '75'
+	} ];
+
+	testUtils.createSinonSandbox();
+
+	beforeEach( () => {
+		editorElement = document.createElement( 'div' );
+		document.body.appendChild( editorElement );
+
+		return ClassicTestEditor
+			.create( editorElement, {
+				plugins: [ Image, ImageStyle, Paragraph, Undo, Table, ImageResizeUI ],
+				image: {
+					resizeUnit: '%',
+					imageResizeOptions: resizeOptions
+				}
+			} )
+			.then( newEditor => {
+				editor = newEditor;
+				command = editor.commands.get( 'imageResize' );
+				plugin = editor.plugins.get( 'ImageResizeUI' );
+			} );
+	} );
+
+	afterEach( () => {
+		if ( editorElement ) {
+			editorElement.remove();
+		}
+
+		if ( editor ) {
+			return editor.destroy();
+		}
+	} );
+
+	describe( 'plugin', () => {
+		it( 'should be named', () => {
+			expect( ImageResizeUI.pluginName ).to.equal( 'ImageResizeUI' );
+		} );
+
+		it( 'should be bound to `imageResize#isEnabled`', () => {
+			command.isEnabled = true;
+
+			expect( plugin.isEnabled ).to.be.true;
+
+			command.isEnabled = false;
+
+			expect( plugin.isEnabled ).to.be.false;
+		} );
+	} );
+
+	describe( 'init()', () => {
+		it( 'should have set "%" resize unit', () => {
+			expect( plugin._resizeUnit ).to.equal( '%' );
+		} );
+
+		it( 'should have set "%" resize unit if not defined', () => {
+			return ClassicTestEditor
+				.create( editorElement, {
+					plugins: [ Image, ImageStyle, Paragraph, Undo, Table, ImageResizeUI ],
+					image: {
+						imageResizeOptions: resizeOptions
+					}
+				} )
+				.then( newEditor => {
+					const plugin = newEditor.plugins.get( 'ImageResizeUI' );
+
+					expect( plugin._resizeUnit ).to.equal( '%' );
+
+					newEditor.destroy();
+				} );
+		} );
+
+		it( 'should have set "px" resize unit', () => {
+			return ClassicTestEditor
+				.create( editorElement, {
+					plugins: [ Image, ImageStyle, Paragraph, Undo, Table, ImageResizeUI ],
+					image: {
+						resizeUnit: 'px',
+						imageResizeOptions: resizeOptions
+					}
+				} )
+				.then( newEditor => {
+					const plugin = newEditor.plugins.get( 'ImageResizeUI' );
+
+					expect( plugin._resizeUnit ).to.equal( 'px' );
+
+					newEditor.destroy();
+				} );
+		} );
+
+		it( 'should have configured resize options', () => {
+			const imageResizeOptions = editor.config.get( 'image.imageResizeOptions' );
+
+			expect( imageResizeOptions.length ).to.equal( 3 );
+		} );
```

Comment:
My intention was to test this line https://github.com/ckeditor/ckeditor5/pull/7482/files#diff-7bb4182967cdad6d262c8cff094900aaR50.

Reply:
In this case the test should check if you create desired number of UI buttons from given configuration. Testing if the configuration was passed to the `editor.config` doesn't bring a value IMO. Tests for that are in https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-utils/tests/config.js.

A feature configuration test should check if it change it's behaviour when the configuration is set (for instance when it is different from a default).

To put this a bit differently, If you change the test a bit it can become:

```javascript
test( '...', async () => {
	const editor = await ClassiTestEditor.create( element, { foo: 'bar' } );
	
	const fooConfig = editor.config.get( 'foo' );
	
	expect( fooConfig ).to.equal( 'bar' );
} );
```

The test above will test if the passed value to `foo` configuration is retain in configuration object. It did not define how the feature behaves or reacts to a configuration.

Reply:
OK, got it 👌